src/backend/expungeservice/expunger/analyzers/time_analyzer.py

```diff
@@ -58,13 +60,39 @@ def _mark_all_acquittals_ineligible_using_mrd_date(expunger):
         for charge in expunger.acquittals:
             charge.set_time_ineligible('Recommend sequential expungement', eligibility_date)
 
+    @staticmethod
+    def _calculate_has_subsequent_charge(date, charges):
+
+        for charge in charges:
+            if charge.acquitted():
+                date_of_charge = charge.date
+            else:
+                date_of_charge = charge.disposition.date
+
+            if date_of_charge > date:
+                return True
+        return False
+
+
     @staticmethod
     def _evaluate_class_b_felonies(expunger):
-        if expunger.most_recent_charge and expunger.most_recent_charge.disposition.date > TimeAnalyzer.TWENTY_YEARS_AGO:
-            for charge in expunger.class_b_felonies:
-                charge.set_time_ineligible('Time-ineligible under 137.225(5)(a)(A)(i)',
-                                           expunger.most_recent_charge.disposition.date + relativedelta(
-                                               years=TimeAnalyzer.TWENTY_YEARS))
+        for class_b_felony in expunger.class_b_felonies:
+            # If the B felony is acquitted, it is affected by the other time eligibility rules,
+            # so leave it untouched here.
+            # If it's convicted, this restriction overrides any of those rules.
+            if not class_b_felony.acquitted():
+                other_charges = [ charge for charge in expunger.charges if charge != class_b_felony]
+                if TimeAnalyzer._calculate_has_subsequent_charge(class_b_felony.disposition.date, other_charges):
```

Comment:
I would prefer if we passed in `_calculate_has_subsequent_charge(class_b_felony, expunger.charges)` as `other_charges` is only relevant in calculating whether there is a subsequent charge or not.